src/cmd2.cc

```diff
@@ -556,58 +545,9 @@ void do_cmd_go_down(void)
 
 		/* Leaving */
 		p_ptr->leaving = TRUE;
-
-		if (!fall_trap)
-		{
```

Comment:
I'm pretty sure this is actually not a problem - the original 'if' check there seems to be redundant, and only would apply w.r.t. traps anyway. I'll see if there's any staircase bugginess, but the code looks okay.